src/main/java/com/powsybl/openloadflow/dc/DcLoadFlowEngine.java

```diff
@@ -222,10 +216,18 @@ public static <T> List<DcLoadFlowResult> run(T network, LfNetworkLoader<T> netwo
     }
 
     /**
-     * A simplified version of DcLoadFlowEngine that supports on the fly bus and branch disabling and that do not
-     * update the state vector and the network at the end (because we don't need it to just evaluate a few equations).
+     * A simplified version of DcLoadFlowEngine that supports on the fly bus and branch disabling.
+     * Note that it does not update the state vector and the network at the end (because we don't need it to just evaluate a few equations).
      */
     public static double[] run(DcLoadFlowContext loadFlowContext, DisabledNetwork disabledNetwork, ReportNode reportNode) {
+        return run(loadFlowContext, disabledNetwork, reportNode, new ArrayList<>());
+    }
+
+    /**
+     * A simplified version of DcLoadFlowEngine that supports on the fly bus and branch disabling, and pst actions.
+     * Note that it does not update the state vector and the network at the end (because we don't need it to just evaluate a few equations).
+     */
+    public static double[] run(DcLoadFlowContext loadFlowContext, DisabledNetwork disabledNetwork, ReportNode reportNode, List<LfAction> lfActions) {
         Collection<LfBus> remainingBuses;
```

Comment:
This method:
    Is Badly Named. It only supports PST actions. That should be reflected
   Is specific to the WoodBuryEngine (because it has limitations on actions) and should not be in this class.

Reply:
Suggestion for refactoring
   Rename run as runWithModifiedTargerVector
   In DCLoadFlowEngine:
        create a an method cloneTargetVector with context, disabledNetwork as argument
        the runWithModifiedTargetVector calls this, then run
   In WoodburyEngine
     create a runWithModifiedTargetVector that takes also the list of supported actions
     call the pritected clonetTargetVector from super
     perform the additional modifications from the actions
     then run

Reply:
In this case I propose to rename both methods `runDcLoadFlowWithModifiedTargetVector`, and move them in `WoodburyEngine` (to be specific to fast DC). Do you agree with that @vidaldid-rte ?

Reply:
After some thought, this is more of a refactoring task than development related to the PST feature. Therefore, it should be handled in a separate PR. I’ve opened #1126 here accordingly (can be handled before or after merging current PR).